app/controllers/dashboard/protocol_merges_controller.rb

```diff
@@ -159,4 +162,21 @@ def role_should_be_assigned?(role_to_be_assigned, protocol)
   def has_research?(protocol, research_type)
     protocol.research_types_info.try(research_type) || false
   end
+
+  # We can not merge if 2 requests are under the same process ssrs org AND either of those requests are locked or incomplete
+  def requests_not_merge_eligible?(master, merged)
+    master_requests = master.sub_service_requests
+    merged_requests = merged.sub_service_requests
+
+    master_requests.each do |master_request|
```

Comment:
I'm wondering if this can be made more efficient because it's currently O(n^2).

Based on the condition for whether two protocols can't be merged, I think we could do this
```
# Grab all locked / incomplete requests from each protocol
master_ineligible_requests = master.sub_service_requests.select{ |ssr| ssr.is_locked? || !ssr.is_complete?}
merged_ineligible_requests = merged.sub_service_requests.select{ |ssr| ssr.is_locked? || !ssr.is_complete?}

# Take the intersection of the potentially ineligible requests to see if any of them have the same "organization_id"
return (master_ineligible_requests.map(&:organization_id) & merged_ineligible_requests.map(&:organization_id)).any?
```

Which would be O(2n)

Reply:
I like it and will make the requested changes.